lib/util/transport.js

```diff
@@ -158,6 +197,27 @@ var transport_decode = function (buffer) {
 
     while (off < buffer.length) {
 
+        if (buffer.slice(0, 10).toString() === "PACKED_XML") {
+            try {
+                ct = packed_xml.unpack(buffer);
+                ct = decode_packed_xml_contents(ct);
+            } catch (e) {
+                // let the user know that something wrong,
+                // as exception will be eaten later...
+                console.error(e.message);
+                throw e;
+            }
+            ret.header = {};
+            for (var doc in ct) {
```

Comment:
NOPE. don't do such things in JS - it will hit back. Also see note at line 161 (use array instead of object)